Makefile

```diff
@@ -87,18 +87,26 @@ venv: $(PY)
 $(JUJUGUI): $(PYRAMID)
 	$(PY) setup.py develop
 
-$(MODULESMIN): $(NODE_MODULES) $(PYRAMID) $(MIN_JS_FILES) $(TEMPLATES_FILE) $(BUILT_YUI) $(BUILT_JS_ASSETS) $(BUILT_D3)
+$(MODULESMIN): $(NODE_MODULES) $(PYRAMID) $(BUILT_RAWJSFILES) $(MIN_JS_FILES) $(TEMPLATES_FILE) $(BUILT_YUI) $(BUILT_JS_ASSETS) $(BUILT_D3)
 	bin/python scripts/generate_modules.py -n YUI_MODULES -s $(GUIBUILD)/app -o $(MODULES) -x "(-min.js)|(\/yui\/)"
 	$(NODE_MODULES)/.bin/uglifyjs --screw-ie8 $(MODULES) -o $(MODULESMIN)
 
 .PHONY: modules-js
 modules-js: $(MODULESMIN)
 
-$(GUIBUILD)/app/%-min.js: $(NODE_MODULES)
-	mkdir -p $(@D)
-	cp -r $(GUISRC)/app/$*.js $(GUIBUILD)/app/$*.js
+# The build-js target is used to build and minify only the js files that have
+# changed since the last time they were built.
+.PHONY: build-js
+build-js: $(BUILT_RAWJSFILES) $(MIN_JS_FILES)
+
+$(GUIBUILD)/app/%-min.js: $(GUIBUILD)/app/%.js
+	@#cp -r $(GUISRC)/app/$*.js $(GUIBUILD)/app/$*.js
 	$(NODE_MODULES)/.bin/uglifyjs --screw-ie8 $(GUISRC)/app/$*.js -o $@
 
+$(GUIBUILD)/app/%.js: $(GUISRC)/app/%.js
```

Comment:
I removed the dependency on NODE_MODULES because when it's written in this way it's depending on that for every file which I felt was a little overkill. I wasn't able to find a way where these could be triggered without another target which didn't already require NODE_MODULES. 

Of course now I see `build-js` doesn't have it, and should :)